src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -16,36 +22,45 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book management", description = "Endpoints for managing books")
 @RestController
 @RequestMapping("/api/books")
 @RequiredArgsConstructor
+@Validated
 public class BookController {
     private final BookService bookService;
-    
+
+    @Operation(summary = "Get all books", description = "Get a list of all books")
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
+    @Operation(summary = "Get book by id", description = "Get a book by id")
     @GetMapping("/{id}")
-    public BookDto getById(@PathVariable Long id) {
+    public BookDto getById(@PathVariable @PositiveOrZero Long id) {
         return bookService.getById(id);
     }
 
+    @Operation(summary = "Delete book by id", description = "Deletes a book by id")
     @ResponseStatus(HttpStatus.NO_CONTENT)
```

Comment:
consider adding response statuses for all other endpoints

Reply:
Other operations have default OK status